core/src/main/scala/kafka/server/ConfigAdminManager.scala

```diff
@@ -499,7 +499,8 @@ object ConfigAdminManager {
             .orElse(Option(ConfigDef.convertToString(configKeys(configPropName).defaultValue, ConfigDef.Type.LIST)))
             .getOrElse("")
             .split(",").toList
-          val newValueList = oldValueList ::: alterConfigOp.configEntry.value.split(",").toList
+          val appendingValueList = alterConfigOp.configEntry.value.split(",").toList.filter(value => !oldValueList.contains(value))
+          val newValueList = oldValueList ::: appendingValueList
```

Comment:
Do you know if there are any cases where the ordering of the list is significant?

Reply:
There are 12 KafkaConfig configs and 3 LogConfig configs with List type:

zookeeper.ssl.enabled.protocols
zookeeper.ssl.cipher.suites
ssl.enabled.protocols
ssl.cipher.suites
sasl.enabled.mechanisms
sasl.kerberos.principal.to.local.rules
sasl.oauthbearer.expected.audience
process.roles
log.cleanup.policy
metric.reporters
kafka.metrics.reporters
controller.quorum.voters

cleanup.policy
leader.replication.throttled.replicas
follower.replication.throttled.replicas

Not sure about the ssl and sasl related configs, the others all don't care about the ordering, but we are still trying to keep their original order now.